syne_tune/blackbox_repository/conversion_scripts/scripts/hpob_import.py

```diff
@@ -0,0 +1,606 @@
+            objectives_names=objectives_names,
+        )
+    return bb_dict
+
+
+def generate_hpob(search_space):
+    print("generating hpob_" + search_space["name"])
+    raw_data_dicts = load_data()
+    merged_datasets = merge_multiple_dicts(
+        raw_data_dicts[0], raw_data_dicts[1], raw_data_dicts[2]
+    )
+    df = pd.DataFrame.from_dict(merged_datasets)
+
+    blackbox_name = BLACKBOX_NAME + search_space["name"]
+    bb_dict = {}
+    indices_not_nan = df[df[search_space["name"]].notna()].index
+    for dataset_name in indices_not_nan:
+        bb_dict[dataset_name] = convert_dataset(
+            search_space, dataset_name, df[search_space["name"]][dataset_name]
+        )
+    with catchtime("saving to disk"):
+        serialize(
+            bb_dict=bb_dict,
+            path=repository_path / blackbox_name,
+            metadata={metric_elapsed_time: METRIC_ELAPSED_TIME},
+        )
+
+
+def convert_dataset(search_space, dataset_name, dataset):
+    hp_cols = list(search_space["config_space"].keys())
+    n_hps = len(hp_cols)
+    n_evals = len(dataset["X"])
+    hps = np.zeros((n_evals, n_hps))
+
+    for i, config in enumerate(dataset["X"]):
+        hps[i] = config
+
+    hyperparameters = pd.DataFrame(data=hps, columns=hp_cols)
+
+    objective_names = ["metric_accuracy", "metric_elapsed_time"]
+
+    objective_evaluations = np.array(dataset["y"])  # np.array.shape = (N,)
+    objective_evaluations = objective_evaluations.reshape(
+        objective_evaluations.shape[0], 1, 1, 1
+    )
+
+    # Create a metric_elapsed_time array filled with ones as runtime was not provided in the dataset
+    elapsed_time_array = np.ones_like(objective_evaluations)
+
+    objective_evaluations = np.concatenate(
+        [objective_evaluations, elapsed_time_array], axis=-1
+    )
+
+    fidelity_space = {
+        RESOURCE_ATTR: randint(lower=MAX_RESOURCE_LEVEL, upper=MAX_RESOURCE_LEVEL)
+    }
+
+    return BlackboxTabular(
+        hyperparameters=hyperparameters,
+        configuration_space=search_space["config_space"],
+        objectives_evaluations=objective_evaluations,
+        objectives_names=objective_names,
+        fidelity_space=fidelity_space,
+    )
+
+
+def load_data():
+    hpob_data_file = repository_path / "hpob-data.zip"
+    if not hpob_data_file.exists():
+        data_src = "https://rewind.tf.uni-freiburg.de/index.php/s/xdrJQPCTNi2zbfL/download/hpob-data.zip"
+        print(f"did not find {hpob_data_file}, downloading {data_src}")
+        urllib.request.urlretrieve(data_src, hpob_data_file)
+
+        with zipfile.ZipFile(hpob_data_file, "r") as zip_ref:
+            zip_ref.extractall()
+
+    meta_test_file = "hpob-data/meta-test-dataset.json"
+    meta_train_file = "hpob-data/meta-train-dataset.json"
+    meta_validation_file = "hpob-data/meta-validation-dataset.json"
+
+    # search spaces: 4796, 5527, 5636, 5859, 5860, 5891, 5906, 5965, 5970, 5971, 6766, 6767, 6794, 7607, 7609, 5889
```

Comment:
Do we need this comment here, since we list the search spaces above? If not let's remove it